keras_cv/models/segmentation/deeplab_v3_plus/deeplab_v3_plus.py

```diff
@@ -0,0 +1,223 @@
+
+        low_level_features = low_level_feature_extractor(inputs)
+        low_level_projected_features = low_level_feature_projector(
+            low_level_features
+        )
+
+        encoder_outputs = keras.layers.UpSampling2D(
+            size=(
+                low_level_projected_features.shape[1] // spp_outputs.shape[1],
+                low_level_projected_features.shape[2] // spp_outputs.shape[2],
+            ),
+            interpolation="bilinear",
+            name="encoder_output_upsampling",
+        )(spp_outputs)
+
+        combined_encoder_outputs = keras.layers.Concatenate(axis=-1)(
+            [encoder_outputs, low_level_projected_features]
+        )
+
+        if segmentation_head is None:
+            segmentation_head = keras.Sequential(
+                [
+                    keras.layers.Conv2D(
+                        name="segmentation_head_conv",
+                        filters=256,
+                        kernel_size=1,
+                        padding="same",
+                        use_bias=False,
+                    ),
+                    keras.layers.BatchNormalization(
+                        name="segmentation_head_norm"
+                    ),
+                    keras.layers.ReLU(name="segmentation_head_relu"),
+                    keras.layers.UpSampling2D(
+                        size=(4, 4), interpolation="bilinear"
+                    ),
+                ]
+            )
+
+            if dropout:
+                segmentation_head.add(
+                    keras.layers.Dropout(
+                        dropout, name="segmentation_head_dropout"
+                    )
+                )
+
+            # Classification layer
+            segmentation_head.add(
+                keras.layers.Conv2D(
+                    name="segmentation_output",
+                    filters=num_classes,
+                    kernel_size=1,
+                    use_bias=False,
+                    padding="same",
+                    activation="softmax",
+                    # Force the dtype of the classification layer to float32
+                    # to avoid the NAN loss issue when used with mixed
+                    # precision API.
+                    dtype=tf.float32,
+                )
+            )
+
+        outputs = segmentation_head(combined_encoder_outputs)
+
+        super().__init__(inputs=inputs, outputs=outputs, **kwargs)
+
+        self.num_classes = num_classes
+        self.backbone = backbone
+        self.spatial_pyramid_pooling = spatial_pyramid_pooling
+        self.projection_filters = projection_filters
+        self.segmentation_head = segmentation_head
+
+    def get_config(self):
+        return {
+            "num_classes": self.num_classes,
+            "backbone": self.backbone,
+            "spatial_pyramid_pooling": self.spatial_pyramid_pooling,
+            "projection_filters": self.projection_filters,
+            "segmentation_head": self.segmentation_head,
```

Comment:
I still struggle with serialization. Do we need to explicitly serialize and deserialize a submodel like this for saving to work properly? I see a saving test but I know there are gotchas.